src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```diff
@@ -304,1237 +234,653 @@ export default class Fms {
-        // TODO: change to switch with a default throw
-        if (category === FLIGHT_CATEGORY.ARRIVAL) {
-            this.setArrivalRunway(runway);
-        } else if (category === FLIGHT_CATEGORY.DEPARTURE) {
-            this.setDepartureRunway(runway);
+            return;
         }
-    }
 
-    /**
-     * Given a `procedureId` find the `#collectionName` that
-     * procedure belongs to, then translate that `#collectionName`
-     * to a `flightPhase`.
-     *
-     * @for Fms
-     * @method _translateProcedureNameToFlightPhase
-     * @param procedureId {string}
-     * @return {string}
-     * @private
-     */
-    _translateProcedureNameToFlightPhase(procedureId) {
-        const collectionToFlightPhaseDictionary = {
-            sidCollection: FLIGHT_CATEGORY.DEPARTURE,
-            starCollection: FLIGHT_CATEGORY.ARRIVAL
-        };
-        const collectionName = this._navigationLibrary.findCollectionNameForProcedureId(procedureId);
-
-        return collectionToFlightPhaseDictionary[collectionName];
+        this.setDepartureRunway(departureRunwayModel);
     }
 
     /**
-     * Removes `LegModel`s from index `0` to `#legIndex`.
-     *
-     * This method is useful for removing a specific number of `LegModel`s
-     * from the left of the collection.
+     * Verify that this FMS's route contains at least two waypoints, or throw an error
      *
-     * @for Fms
-     * @method _trimLegCollectionAtIndex
-     * @param legIndex {number}
-     * @private
-     */
-    _trimLegCollectionAtIndex(legIndex) {
-        this.legCollection = this.legCollection.slice(legIndex);
-    }
-
-    // TODO: simplify this and abstract it away from `.prependLeg()`
-    /**
-     * Given an array of `routeSegments`, prepend each to the left of the `#legCollection`
+     * This is an expectation on spawn so we can point the aircraft somewhere,
+     * but OTHER than on spawn, there is no reason an aircraft cannot be assigned
+     * a single-fix route (such as "forget everything else, just go direct to the airport!").
      *
      * @for Fms
-     * @method _prependLegCollectionWithRouteAmendment
-     * @param routeSegments {array<string>}             direct or procedure routeStrings
+     * @method _verifyRouteContainsMultipleWaypoints
      * @private
      */
-    _prependLegCollectionWithRouteAmendment(routeSegments) {
-        // reversing order here because we're leveraging `.prependLeg()`, which adds a single
-        // leg to the left of the `#legCollection`. by reversing the array, we can ensure the
-        // correct order of legs.
-        const routeSegmentList = routeSegments.slice().reverse();
-
-        for (let i = 0; i < routeSegmentList.length; i++) {
-            const routeSegment = routeSegmentList[i];
-            const legModel = this._buildLegModelFromRouteSegment(routeSegment);
-
-            this.prependLeg(legModel);
+    _verifyRouteContainsMultipleWaypoints() {
+        if (this.waypoints.length < 2) {
```

Comment:
reverse logic and return early for valid cases, then handle the hidden `else` afterwards with the `throw`